src/bci_build/package/appcontainers.py

```diff
@@ -147,36 +147,28 @@
 MARIADB_CONTAINERS = []
 MARIADB_CLIENT_CONTAINERS = []
 
-for os_version in set(ALL_NONBASE_OS_VERSIONS) | {OsVersion.BASALT}:
+for mariadb_version, os_version in zip(
+    ("10.6", "10.11", "11.2", "10.11"), ALL_NONBASE_OS_VERSIONS + [OsVersion.BASALT]
+):
     if os_version in (OsVersion.BASALT, OsVersion.TUMBLEWEED):
         prefix = ""
         additional_names = []
     else:
         prefix = "rmt-"
         additional_names = ["mariadb"]
 
+    version_check_lines = f"""# sanity check that the version from the tag is equal to the version that we expect
+{DOCKERFILE_RUN} [[ $(rpm -q --qf "%{{version}}" mariadb-client | cut -d "." -f -2) = "{mariadb_version}" ]]"""
+
     MARIADB_CONTAINERS.append(
         ApplicationStackContainer(
             package_name=f"{prefix}mariadb-image",
             additional_names=additional_names,
             os_version=os_version,
             is_latest=os_version in CAN_BE_LATEST_OS_VERSION,
             name=f"{prefix}mariadb",
-            version="%%mariadb_version%%",
+            version=mariadb_version,
```

Comment:
Possibly all containers use a variation of `%%{name}_version%%`, and I don't see this hardcoding scaling well.

It would be easier to remove `%%{name}_version%%` references from our READMES and just use `registry.suse.com/suse/{image}:latest`.

The README is just an example, it does not need to show the correct version IMHO, all containers should have a "latest version".